Operator container panic because of invalid pod state

**I'm submitting a bug**

 <!--- What kind of an issue is this? Put an `x` in all the boxes that apply: -->
 - [x] bug report
 - [ ] feature request
 - [ ] support request



**Do you want to request a *feature* or report a *bug*?**
Report a bug in https://github.com/CrunchyData/postgres-operator/blob/master/internal/operator/cluster/clusterlogic.go#L638


**What is the current behavior?**
Operator pod crashes because of incorrect kubernetes response handling when no pods are present while attempting to shutdown a cluster


**If the current behavior is a bug, please provide the steps to reproduce:**
- Have a PGCluster which is to be shut down with no pods present
- Operator container crashes because of `panic`


**What is the expected behavior?**
Operator skips PGCluster as no pods are to be shutdown, or handles the exception without the container exiting.


**What is the motivation or use case for changing the behavior?**
-


**Other information** (e.g. detailed explanation, related issues, suggestions how to fix, etc)
```
time="2020-11-26T12:55:56Z" level=debug msg="pgcluster onUpdate for cluster ******* (namespace ********)" func="internal/controller/pgcluster.(*Controller).onUpdate()" file="internal/controller/pgcluster/pgclustercontroller.go:195" version=4.5.0
E1126 12:55:56.810118       1 runtime.go:78] Observed a panic: runtime.boundsError{x:0, y:0, signed:true, code:0x0} (runtime error: index out of range [0] with length 0)
goroutine 1980 [running]:
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1614080, 0xc00270c220)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1614080, 0xc00270c220)
    /usr/lib/golang/src/runtime/panic.go:679 +0x1b2
github.com/crunchydata/postgres-operator/internal/operator/cluster.ShutdownCluster(0x1a39320, 0xc000a85780, 0x0, 0x0, 0x0, 0x0, 0xc0028e55c0, 0x8, 0x0, 0x0, ...)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/internal/operator/cluster/clusterlogic.go:628 +0xef2
github.com/crunchydata/postgres-operator/internal/controller/pgcluster.(*Controller).onUpdate(0xc000a83050, 0x16b68e0, 0xc000cb7100, 0x16b68e0, 0xc0026a8700)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/internal/controller/pgcluster/pgclustercontroller.go:210 +0x1451
github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnUpdate(...)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache/controller.go:205
github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1.1(0xc00022e700, 0x7f62be2d4008, 0x0)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache/shared_informer.go:656 +0x188
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0xc00277bdd8, 0x1, 0xc000db9c00)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:292 +0x51
github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1()
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache/shared_informer.go:652 +0x79
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0014adf40)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00277bf40, 0xdf8475800, 0x0, 0x42ee01, 0xc00063cde0)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until(...)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache.(*processorListener).run(0xc000a98a80)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache/shared_informer.go:650 +0x9b
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc000793d40, 0xc000548c20)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x59
created by github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:69 +0x62
panic: runtime error: index out of range [0] with length 0 [recovered]
    panic: runtime error: index out of range [0] with length 0
goroutine 1980 [running]:
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x1614080, 0xc00270c220)
    /usr/lib/golang/src/runtime/panic.go:679 +0x1b2
github.com/crunchydata/postgres-operator/internal/operator/cluster.ShutdownCluster(0x1a39320, 0xc000a85780, 0x0, 0x0, 0x0, 0x0, 0xc0028e55c0, 0x8, 0x0, 0x0, ...)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/internal/operator/cluster/clusterlogic.go:628 +0xef2
github.com/crunchydata/postgres-operator/internal/controller/pgcluster.(*Controller).onUpdate(0xc000a83050, 0x16b68e0, 0xc000cb7100, 0x16b68e0, 0xc0026a8700)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/internal/controller/pgcluster/pgclustercontroller.go:210 +0x1451
github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnUpdate(...)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache/controller.go:205
github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1.1(0xc00022e700, 0x7f62be2d4008, 0x0)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache/shared_informer.go:656 +0x188
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0xc00277bdd8, 0x1, 0xc000db9c00)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:292 +0x51
github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache.(*processorListener).run.func1()
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache/shared_informer.go:652 +0x79
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0014adf40)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00277bf40, 0xdf8475800, 0x0, 0x42ee01, 0xc00063cde0)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until(...)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache.(*processorListener).run(0xc000a98a80)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/client-go/tools/cache/shared_informer.go:650 +0x9b
github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc000793d40, 0xc000548c20)
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x59
created by github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start
    /opt/cdev/src/github.com/crunchydata/postgres-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:69 +0x62
```


**Please tell us about your environment:**

 - Operating System: CentOS
 - Container Image Tag: b8fd203a4f458acc0d3103417afc47da8810eb70f1a0064419fb28c79e91caba
 - Operator Version: 4.5.0
 - Storage (NFS, hostpath, storage class): storage class
 - PostgreSQL Version: ~
 - Platform (Docker, Kubernetes, OpenShift): Kubernetes
 - Platform Version: v1.18.8

--- a/internal/operator/cluster/clusterlogic.go
+++ b/internal/operator/cluster/clusterlogic.go
@@ -630,7 +630,10 @@ func ShutdownCluster(clientset kubeapi.Interface, cluster crv1.Pgcluster) error 
 		return err
 	}
 
-	if len(pods.Items) > 1 {
+	if len(pods.Items) == 0 {
+		return fmt.Errorf("Cluster Operator: Could not find primary pod for shutdown of "+
+			"cluster %s", cluster.Name)
+	} else if len(pods.Items) > 1 {
 		return fmt.Errorf("Cluster Operator: Invalid number of primary pods (%d) found when "+
 			"shutting down cluster %s", len(pods.Items), cluster.Name)
 	}

